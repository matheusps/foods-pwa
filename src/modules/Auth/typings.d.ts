interface IAuthentication {
  handleAuthentication(): void
  getAccessToken(): any
  getIdToken(): any
  setSession(authResult: Auth0DecodedHash): void
  renewSession(): void
  login(): void
  logout(): void
  isAuthenticated(): boolean
}

interface IAuthConfig {
  domain: string
  clientID: string
  redirectUri: string
  responseType: string
  scope: string
}

interface IAuthContext {
  auth: IAuthentication
}
