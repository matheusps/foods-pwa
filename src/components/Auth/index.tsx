import history from '../../history'
import auth0, { Auth0DecodedHash, WebAuth } from 'auth0-js'

export default class Auth {
  private accessToken: any = ''
  private idToken: any = ''
  private expiresAt: any = 0

  private auth0: WebAuth = new auth0.WebAuth({
    domain: 'matheusps.auth0.com',
    clientID: 'kvdH3Y0PQDugcCS7k3iJucQxYnM3JSKF',
    redirectUri: 'http://localhost:3000/callback',
    responseType: 'token id_token',
    scope: 'openid',
  })

  handleAuthentication = () =>
    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {
        this.setSession(authResult)
      } else if (err) {
        history.replace('/home')
        console.log(err)
        alert(`Error: ${err.error}. Check the console for further details.`)
      }
    })

  getAccessToken = () => this.accessToken

  getIdToken = () => this.idToken

  setSession = (authResult: Auth0DecodedHash) => {
    localStorage.setItem('isLoggedIn', 'true')
    this.accessToken = authResult.accessToken
    this.idToken = authResult.idToken
    this.expiresAt = authResult.expiresIn! * 1000 + new Date().getTime()
    history.replace('/home')
  }

  renewSession = () =>
    this.auth0.checkSession({}, (err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {
        this.setSession(authResult)
      } else if (err) {
        this.logout()
        console.log(err)
        alert(
          `Could not get a new token (${err.error}: ${err.errorDescription}).`
        )
      }
    })

  login = () => {
    this.auth0.authorize()
  }

  logout = () => {
    this.accessToken = ''
    this.idToken = ''
    this.expiresAt = 0
    localStorage.removeItem('isLoggedIn')
    history.replace('/home')
  }

  isAuthenticated = () => new Date().getTime() < this.expiresAt
}
