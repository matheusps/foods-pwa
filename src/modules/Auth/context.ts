import { createContext } from 'react'

export interface IAuthContext {
  isAuth: boolean
  auth: IAuthentication
}

const AuthContext = createContext<IAuthContext>(null!)
const { Provider, Consumer } = AuthContext

export default AuthContext
export { Provider as AuthProvider, Consumer as AuthConsumer }
