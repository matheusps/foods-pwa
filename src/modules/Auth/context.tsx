import React, { createContext } from 'react'
import AuthService from './service'

export const AuthContext = createContext<IAuthContext>(null!)
const { Provider, Consumer } = AuthContext

const AuthProvider = ({ children }: any) => (
  <Provider value={{ auth: AuthService }}>{children}</Provider>
)

export default AuthContext
export { AuthProvider, Consumer as AuthConsumer }
