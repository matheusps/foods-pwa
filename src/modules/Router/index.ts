import history from './history'
import HistoryContext, { HistoryProvider, HistoryConsumer } from './context'
import useLocation from './hooks/useLocation'
import useRouter from './hooks/useRouter'
import Link from './components/Link'
import Router from './components/Router'

export {
  history,
  HistoryContext,
  HistoryProvider,
  HistoryConsumer,
  useLocation,
  useRouter,
  Link,
  Router,
}
