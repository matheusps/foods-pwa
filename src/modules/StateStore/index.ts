import {
  createContext,
  createElement,
  useState,
  useEffect,
  useContext,
  Dispatch,
} from 'react'

import { List } from 'immutable'

const createStore: CreateStore = (reducer: Reducer, initialState) => {
  let store = initialState
  let subscribers = List<Dispatch<any>>()

  const Context = createContext(initialState)

  const StoreProvider = ({ children }: StoreProviderProps) => {
    const [state, setState] = useState(initialState)
    useEffect(() => {
      subscribers = subscribers.push(setState)
      store !== initialState && setState(store)
      return () => {
        subscribers = subscribers.delete(subscribers.indexOf(setState))
      }
    }, [store])
    return createElement(Context.Provider, { value: state }, children)
  }

  const useStore: UseStore = (name: string) => {
    const state = useContext(Context)
    return state[name]
  }

  const setState: StateMutation = state => {
    store = state
    subscribers.forEach(f => f(store))
  }

  const dispatch: Dispatch<any> = action => {
    const newState = reducer(store, action)
    setState(newState)
    return action
  }

  return {
    StoreProvider,
    useStore,
    store,
    dispatch,
  }
}

export default createStore
