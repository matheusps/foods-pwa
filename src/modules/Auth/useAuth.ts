import { useContext } from 'react'
import { AuthContext } from './context'

const useAuth = () => {
  const { auth } = useContext(AuthContext)
  return {
    ...auth,
  }
}

export default useAuth
