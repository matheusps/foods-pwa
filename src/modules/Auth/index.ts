import AuthContext, { AuthProvider, AuthConsumer } from './context'
import AuthService from './service'
import useAuth from './useAuth'

export { AuthContext, AuthProvider, AuthConsumer, AuthService, useAuth }
