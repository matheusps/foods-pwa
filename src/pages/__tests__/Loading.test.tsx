import React from 'react'
import { render } from 'react-testing-library'
import { AuthProvider } from '../../modules/Auth'
import LoadingPage from '../../pages/Loading'

describe('Loading Page', () => {
  it('should match snapshot', () => {
    const component = render(
      <AuthProvider>
        <LoadingPage />
      </AuthProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
