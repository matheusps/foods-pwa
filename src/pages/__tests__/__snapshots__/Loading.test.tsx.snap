// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Page should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="justify-center items-center flex w-100 vh-100 sc-bwzfXH jaNCXe pa3"
      >
         
        <div
          class="sc-bdVaJa gddndU"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-center items-center flex w-100 vh-100 sc-bwzfXH jaNCXe pa3"
    >
       
      <div
        class="sc-bdVaJa gddndU"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
